Sources/App/Models/SocialAccount/SocialAccount.swift

```diff
@@ -0,0 +1,78 @@
+import Vapor
+import FluentProvider
+import HTTP
+
+final class SocialAccount: Model {
+  let storage = Storage()
+  
+  var socialId: String
+  var socialUserId: String
+  var userId: Int
+  
+  struct Keys {
+    static let id = "id"
+    static let socialId = "social_id"
+    static let socialUserId  = "social_user_id"
+    static let userId = "user_id"
+  }
+  
+  init(socialId: String, socialUserId: String, userId: Int) {
+    self.socialId = socialId
+    self.socialUserId = socialUserId
+    self.userId = userId
+  }
+  
+  init(row: Row) throws {
+    socialId = try row.get(SocialAccount.Keys.socialId)
+    socialUserId = try row.get(SocialAccount.Keys.socialUserId)
+    userId = try row.get(SocialAccount.Keys.userId)
+    id = try row.get(SocialAccount.Keys.id)
+  }
+  
+  func makeRow() throws -> Row {
+    var row = Row()
+    try row.set(SocialAccount.Keys.id, id)
+    try row.set(SocialAccount.Keys.socialId, socialId)
+    try row.set(SocialAccount.Keys.socialUserId, socialUserId)
+    try row.set(SocialAccount.Keys.userId, userId)
+    return row
+  }
+}
+
+// MARK: Fluent Preparation
+
+extension SocialAccount: Preparation {
+  static func prepare(_ database: Database) throws {
+    try database.create(self) { builder in
+      builder.id()
+      builder.foreignKey(SocialAccount.Keys.socialId, references: Social.Keys.id, on: Social.self, named: "social")
```

Comment:
Let's move out constant from here?

Reply:
To the Keys

Reply:
@smejl there's a better option. Since it a table name key, `Entity` protocol(that `Model` conforms to) has a convenient property `static var entity: String = "social"` that we can override in our models(let's add it on top of model under `storage` property). I've seen this in other Vapor backend examples. So we can refer it with `Social.entity`. 

By the way, i'm not sure, but if we do this, `named` method parameter could be inferred automatically, so no need to include it in method call.(we should check it later)

Reply:
@dannydaddy3 @smejl Entity protocol also has property `name`, which provides the singular relational name of a model (`entity` - the plural relational name of this model). And we can actually ignore `named` parameter in init.
Also there is a `foreignIdKey` property, which we can use for `references` parameter.
So I suggest ignore `named` and use `foreignIdKey`.